src/Compilers/CSharp/Portable/Symbols/LocalSymbol.cs

```diff
@@ -25,6 +25,11 @@ protected LocalSymbol()
             get;
         }
 
+        /// <summary>
+        /// Syntax node that is used as the scope designator. Otherwise, null.
+        /// </summary>
+        internal abstract SyntaxNode ScopeDesignatorOpt { get; }
```

Comment:
The name is a bit misleading. It mostly just for tagging locals into special groups. It seems we have got another kind which is tagged via ScopeDesignatorOpt.  Since we do not seem to be really using the actual syntax except for checking its kind, perhaps just adding one more SynthesizedLocalKind would be sufficient.

---
In reply to: [96762026](https://github.com/dotnet/roslyn/pull/16593#discussion_r96762026) [](ancestors = 96762026)